demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -71,6 +71,14 @@ extern void vLoggingPrint( const char * pcFormat );
  * a new default key pair, regardless of whether an existing key pair is present. */
 #define keyprovisioningFORCE_GENERATE_NEW_KEY_PAIR    0
 
+/* Delay before generating new key-pair, if keyprovisioningFORCE_GENERATE_NEW_KEY_PAIR
+ * is enabled. This is to avoid possible race-condition (due to devce reset) between
+ * execution of an existing image on device generates key-pair on device and flashing of
+ * new image on device. */
+#ifndef keyprovisioningDELAY_BEFORE_KEY_PAIR_GENERATION_SECS
+    #define keyprovisioningDELAY_BEFORE_KEY_PAIR_GENERATION_SECS    30
```

Comment:
so by default, there will be a delay? this will change the behavior of current customers who use this?

Reply:
This is configurable, so users that do not want the delay can change the value to 0 seconds. This information will be provided in the README